packages/opentelemetry-tracing/src/Span.ts

```diff
@@ -171,6 +171,14 @@ export class Span implements api.Span, ReadableSpan {
       );
     }
 
+    if (this.resource instanceof Promise) {
+      this.resource.then(resource => {
```

Comment:
Can we race this against a timeout promise so that we can ensure a slow resource detector does not delay or block tracing?

Reply:
fixes

Reply:
Where is the fix I don't see it?